### PR TITLE
Add some PR templates with checklist to support branching strategy

### PR DESCRIPTION
**Pull request by azvanderpas**
Adds two PR templates for pushing to develop and to main and a template to help selecting the correct template. 
Tested it on my own fork since otherwise its a chicken and egg problem (need to be committed to develop to see the functionality on Github). 

Unsure if it'll still work for develop to main because main is not the 'default' branch.

Resources: 
- https://stackoverflow.com/questions/46912624/is-it-possible-to-have-use-multiple-github-issue-templates-on-github
- https://github.com/EttusResearch/uhd/blob/master/.github/pull_request_template.md?plain=1
- https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/develop.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Pull Request Details
+<!--- The title of the PR should summarize the change implemented. -->
+
+## Description
+<!--- Provide a general summary of your changes in the title above -->
+<!--- Why is this change required? What problem does it solve? -->
+
+## Related Issue
+<!--- Refer to any related issues here -->
+<!--- If this PR fully addresses an issue, please say "Fixes #1234", -->
```
Comment (caseyjbrotherton) [inline .github/PULL_REQUEST_TEMPLATE/develop.md:10]: Are we using github issues or trello cards?

File: .github/PULL_REQUEST_TEMPLATE/develop.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+## Checklist
+
+<!--- Go over all the following points, and put an `x` in all the
+<!--- boxes that apply. Note that some of these may not be valid -->
+<!--- for all PRs. -->
+
+- [ ] My code follows the code style of this project.
+- [ ] My code has been tested on the Robot OR includes simulation code to prove functionality.
+- [ ] I have updated the documentation accordingly.
+- [ ] I have added tests to cover my changes, and all previous tests pass.
```
Comment (caseyjbrotherton) [inline .github/PULL_REQUEST_TEMPLATE/develop.md:33]: We don't use documentation now, and don't have a 100% test case coverage.  We already ask for team members to describe tests above.  Should we remove these two items from the checklist?

**Review by caseyjbrotherton — DISMISSED** [review]
How do we test this?  Merge and test in place?

File: .github/PULL_REQUEST_TEMPLATE/develop.md (hunk 1 above)
Comment (azvanderpas) [reply .github/PULL_REQUEST_TEMPLATE/develop.md:10]: I think we were doing bugs on GH, but trello for work items.

**Comment by azvanderpas** [comment]
> How do we test this? Merge and test in place?

Seems like it. I forked the repo and tested it there and it appears to work though. Selecting the right template is a bit kludgy but works.

**Review by caseyjbrotherton — APPROVED** [review]
Looks good

**Pull request merged**